.changeset/pink-turkeys-live.md

```diff
@@ -0,0 +1,5 @@
+---
+'@urql/exchange-graphcache': patch
+---
+
+Fix issues with continuously updating operations (i.e. subscriptions and `hasNext: true` queries) by shifting their layers in Graphcache behind those that are still awaiting a result. This causes continuous updates to not overwrite one-off query results while still keeping continuously updating operations at the highest possible layer.
```

Comment:
I'm a little unsure of whether to mark this as a fix. Technically, I'd say this is a feature, since previously we knew the lack of this could cause issues but underestimated the nature of concurrent subscription updates.